gnatsd/check.py

```diff
@@ -0,0 +1,150 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import requests
+
+# stdlib
+
+# project
+from checks import AgentCheck
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'gnatsd'
+
+class GnatsdCheck(AgentCheck):
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self.counts = {}
+
+    def check(self, instance):
+        self.InstanceCheck(instance, self).check()
+
+    class InstanceCheck:
+        SERVICE_CHECK_NAME = 'gnatsd.can_connect'
+
+        METRICS = {
+            'varz': {
+                'connections': 'gauge',
+                'subscriptions': 'gauge',
+                'slow_consumers': 'count',
+                'remotes': 'gauge',
+                'routes': 'gauge',
+                'in_msgs': 'count',
+                'out_msgs': 'count',
+                'in_bytes': 'count',
+                'out_bytes': 'count',
+                'mem': 'gauge'
+            },
+            'connz': {
+                'num_connections': 'gauge',
+                'total': 'count',
+                'connections': {
+                    'pending_bytes': 'gauge',
+                    'in_msgs': 'count',
+                    'out_msgs': 'count',
+                    'subscriptions': 'gauge',
+                    'in_bytes': 'count',
+                    'out_bytes': 'count'
+                }
+            },
+            'routez': {
+                'num_routes': 'gauge',
+                'routes': {
+                    'pending_size': 'gauge',
+                    'in_msgs': 'count',
+                    'out_msgs': 'count',
+                    'subscriptions': 'gauge',
+                    'in_bytes': 'count',
+                    'out_bytes': 'count'
+                }
+            }
+        }
+
+        TAGS = {
+            'varz': ['server_id'],
+            'connz.connections': ['cid','ip','name','lang','version'],
+            'routez.routes': ['rid','remote_id','ip']
+        }
+
+        def __init__(self, instance, checker):
+            self.instance = instance
+            self.checker = checker
+            self.config = self.Config(instance)
+            self.tags = self.config.tags + ['server_name:%s' % self.config.server_name]
+            self.service_check_tags = self.tags + ['url:%s' % self.config.host]
+
+        def check(self):
+            # Confirm monitor endpoint is available
+            self._status_check()
+
+            # Gather NATS metrics
+            for endpoint, metrics in self.METRICS.items():
```

Comment:
This is making a lot of api calls. Can you get this data in fewer calls? It's fine if you can't, no big deal. What we have to do we have to do. Just thought I'd ask :)

Reply:
It should only be three API calls - each NATS monitoring endpoint includes different metrics - but as far as I am aware there's no _single_ API endpoint that returns them all.

Reply:
That's fine! Just making sure!